apps/dashboard/src/components/amount-input.tsx

```diff
@@ -24,22 +25,17 @@ export const AmountInput = ({
   className,
   placeholder,
   isReadOnly,
+  onValueChange,
 }: InputWithSelectProps) => {
-  const { getFieldState, setValue, getValues, control } = useFormContext();
+  const { getFieldState, setValue, control } = useFormContext();
 
   const input = getFieldState(`${fields.inputKey}`);
   const select = getFieldState(`${fields.selectKey}`);
   const error = input.error || select.error;
 
-  const handleChange = (value: { input: number; select: string }) => {
-    // we want to always set both values and treat it as a single input
-    setValue(fields.inputKey, value.input, { shouldDirty: true });
-    setValue(fields.selectKey, value.select, { shouldDirty: true });
-  };
-
   return (
     <>
-      <InputFieldPure className="h-7 rounded-lg border pr-0">
+      <InputFieldPure className={cn('h-7 w-auto rounded-lg border pr-0', className)}>
```

Comment:
The `className` should be applied to the parent, otherwise, we can't adjust it to the UI requirements in the digest where the input and the field don't take a full width.